pimd/pim_join.c

```diff
@@ -72,8 +72,21 @@ static void recv_join(struct interface *ifp, struct pim_neighbor *neigh,
 	 * If the RPT and WC are set it's a (*,G)
 	 * and the source is the RP
 	 */
-	if ((source_flags & PIM_RPT_BIT_MASK)
-	    && (source_flags & PIM_WILDCARD_BIT_MASK)) {
+	if (source_flags & PIM_WILDCARD_BIT_MASK) {
+		/* As per RFC 7761 Section 4.9.1:
+		 * The RPT (or Rendezvous Point Tree) bit is a 1-bit value for
+		 * use with PIM Join/Prune messages (see Section 4.9.5.1). If
+		 * the WC bit is 1, the RPT bit MUST be 1.
+		 */
+		if (!(source_flags & PIM_RPT_BIT_MASK)) {
```

Comment:
Can we use CHECK_FLAG() macro for readability everywhere? (I know it's not like this PR's scope, but since we are here...)

Reply:
Since you are asking, I have fixed it 😄